src/Xdl.Internship.Core/Xdl.Internship.Core.DataAccess.MongoDB/Repositories/MongoRepository.cs

```diff
@@ -0,0 +1,120 @@
+﻿using MongoDB.Bson;
+using MongoDB.Driver;
+using System;
+using System.Collections.Generic;
+using System.Linq.Expressions;
+using System.Threading;
+using System.Threading.Tasks;
+using Xdl.Internship.Core.DataAccess.MongoDB.Settings;
+using Xdl.Internship.Core.Models.MongoDB;
+
+namespace Xdl.Internship.Core.DataAccess.MongoDB.Repositories
+{
+    public class MongoRepository<TDocument> : IMongoRepository<TDocument> where TDocument : IModelBase
+    {
+        private readonly Lazy<IMongoCollection<TDocument>> collectionAccessor;
```

Comment:
why repository has responsibility to work with client and collection? why not to work ONLY with collection?